### DISCUSSION TITLE
[GNN-Explainer]  Model can't be interpreted

### DISCUSSION THREAD

**Discussion by zju-wyh** [pyg-team/pytorch_geometric#6048] [Q&A]
I create a model including CNN and GAT,but it can't be interpreted using GNN-Explainer.Can it only explain the GNN model not including CNN and Linear？Looking forward to your answer.
![G$6D` )% TNTD`HMB$Y(85B](https://user-images.githubusercontent.com/103970426/203688562-0cd5d730-08a0-4bff-8037-3fccfb7bf846.png)
![EETKR}ZJJ10OQI8~)A SZTD](https://user-images.githubusercontent.com/103970426/203688571-c82d8808-9aa0-42b8-b594-4b34014f5784.png)

**Comment by rusty1s** [comment]
To the best of my knowledge that should work as long as you have a single message passing layer in your network. Do you have a minimal example to reproduce your issue?

**Reply by zju-wyh** [reply]
Thanks for your answer. Model only with GNN is ok, but adding convolutional layer and Linear layer will report an error:TypeError: forward() got an unexpected keyword argument 'x'.

This is ok
![TMBH{VIPG8~IG6EX2Z95K@6](https://user-images.githubusercontent.com/103970426/203782961-699bbf38-b3f5-4d94-ac71-9c5172fa5b60.png)
This is my model
![GB$7 ${U9N0NS8KX$@4(UMF](https://user-images.githubusercontent.com/103970426/203782982-0110f2af-5cbd-48c9-a922-f7c5b6e300f8.png)
![FA) VLN `K@DXS)ZBQJ`EIF](https://user-images.githubusercontent.com/103970426/203783006-e0c4e01b-60c3-488b-a74b-aa5e97c08f6e.png)

**Reply by rusty1s** [reply]
This is mainly because your `forward` function should not expect `data` as input but rather `x` and `edge_index` (and any additional arguments).

**Reply by HeVLF** [reply]
Hey @rusty1s 
Can my `forward` function receive `x`, `edge_index`, and `edge_attr` or, for the GNNExplainer to work, the `forward` function should only receive `x` and `edge_index`?

**Reply by rusty1s** [reply]
The `GNNExplainer` can take in any arguments that the `forward` pass requests. It is just that `x` and `edge_index` needs to be part of it in order to correctly set up the gradient computation.
